packages/@lwc/rollup-plugin/src/__tests__/index.spec.js

```diff
@@ -109,6 +110,20 @@ describe('javascript relative import', () => {
     });
 });
 
+describe('multi-package-version', () => {
```

Comment:
I think you're missing a test like this:

module a:

lwc.config.json
```json
{
 "modules": [{ "npm": "module-b" }]
}
```

module b:

lwc.config.json
```json
{
  "modules": [{"npm": "module-c"}]
}
```
 
module c:

lwc.config.json:
```json
{
  "modules": [{"dir": "src/modules"}]
}
```

basically module a -> module b -> module c all through lwc.config,json 

Modules from a, b and c should be collected.